File: mobile-client/android/UseCase/Order/Detail/LoadOrderDetailUseCase.kt
package app.mobiling.client.client.usecase.order.detail

import app.mobiling.client.client.contract.order.detail.OrderDetailPayload
import app.mobiling.client.client.data.order.detail.OrderDetailGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class LoadOrderDetailUseCase(
    private val gateway: OrderDetailGateway,
) {
    suspend operator fun invoke(orderId: String): OrderDetailPayload = gateway.loadOrderDetail(orderId)
}

File: client/ios/UseCase/Order/Detail/LoadOrderDetailUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
struct LoadOrderDetailUseCase {
    let gateway: OrderDetailGateway

    func callAsFunction(orderId: String) async throws -> OrderDetailPayload {
        try await gateway.loadOrderDetail(orderId: orderId)
    }
}
